### PR TITLE
mq-hateos - implementation for program

### PR DESCRIPTION
**Pull request by MirlindQeriqi**

### CODE REVIEW

File: src/main/java/org/prnhs/javaee/swim/dto/ProgramDto.java [hunk 1]
```diff
@@ -2,22 +2,24 @@
...
 import java.util.Objects;
+import org.springframework.hateoas.ResourceSupport;
 
 @ApiModel(value="ProgramDto", description="Model for representing a Program")
-public class ProgramDto {
+public class ProgramDto extends ResourceSupport implements Serializable{
     
     @ApiModelProperty(value = "Program's key/id", readOnly = true)
-    private Integer id;
+    private Integer keyId;
```
Comment (mosgjig) [inline src/main/java/org/prnhs/javaee/swim/dto/ProgramDto.java]: just key not keyId

File: src/main/java/org/prnhs/javaee/swim/web/ProgramController.java [hunk 2]
```diff
@@ -53,7 +64,13 @@ public ProgramDto getById(@PathVariable Integer id){
     @ApiResponse(code = 200, message ="OK", response = ProgramDto.class)})
     public List<ProgramDto> getAll(){
         LOGGER.debug("getAll method called in ProgramController class");
-        return programService.getAll();
+        
+        List<ProgramDto> programs = programService.getAll();
+        programs.stream().forEach((program) -> {
```
Comment (mosgjig) [inline src/main/java/org/prnhs/javaee/swim/web/ProgramController.java:69]: you don't need the brackets

**Review by mosgjig — CHANGES_REQUESTED** [review]

File: src/main/java/org/prnhs/javaee/swim/web/ProgramController.java (hunk 2 above)
Comment (MirlindQeriqi) [reply src/main/java/org/prnhs/javaee/swim/web/ProgramController.java:69]: i wrote a for each loop to iterate, and when i switched to stream with netbeans, it added those brackets by itself

Comment (mosgjig) [reply src/main/java/org/prnhs/javaee/swim/web/ProgramController.java:69]: it's a styling standard, but not necessary

**Review by mosgjig — APPROVED** [review]

**Review by mosgjig — APPROVED** [review]

**Pull request merged**